Services/DeviceMethods.cs

```diff
@@ -50,6 +50,8 @@ public class DeviceMethods : IDeviceMethods
             ISmartDictionary deviceState,
             ISmartDictionary deviceProperties)
         {
+            if (methods == null) return;
```

Comment:
Consider logging an error when there are no methods passed.

Reply:
we allow device models without methods